packages/core/src/node/initRsbuild.ts

```diff
@@ -340,6 +340,7 @@ async function createInternalBuildConfig(
         if (chain.plugins.has(CHAIN_ID.PLUGIN.REACT_FAST_REFRESH)) {
           chain.plugin(CHAIN_ID.PLUGIN.REACT_FAST_REFRESH).tap(options => {
             options[0] ??= {};
+            options[0].test = [/\.([cm]js|[jt]sx?|flow)$/i, MDX_OR_MD_REGEXP];
```

Comment:
I didn't realize https://github.com/web-infra-dev/rsbuild/pull/5969 would break HMR in Rspress 😂 Maybe we could add an HMR test case for Rspress HMR